modules/store.js

```diff
@@ -75,6 +75,15 @@ export const reducer = handleActions({
 
 }, initialState);
 
+export const reducer = function wrappedReducer(state, action) {
+  // if state is undefined then it can't be converted to mutable
+  let mutableState = state;
+  if (mutableState !== undefined) {
+    mutableState = getMutableState(state);
+  }
+  return getImmutableState(reduxAsyncReducer(mutableState, action));
+};
+
```

Comment:
this part seem a bit unclear. We have 2 invocations of functions, state is actually previous state, etc.
Could you clarify code a bit so it's more readable?